### PR TITLE
Support batch delete syntax in fe and add pass the delete infomation to be

### PR DESCRIPTION
**Pull request by yangzhg**
## Proposed changes

Part of Issues #4051
Add batch delete synatx support, support the syntax described in  Issues #4051 
Bring the delete infomation to be

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on [Doris's issues](https://github.com/apache/incubator-doris/issues), and have described the bug/feature there in detail
- [x] Commit messages in my PR start with the related issues ID
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I have updated the document
- [x] Any dependent changes have been merged

### CODE REVIEW

File: be/src/exec/broker_scan_node.cpp [hunk 1]
```diff
@@ -71,6 +71,18 @@ Status BrokerScanNode::init(const TPlanNode& tnode, RuntimeState* state) {
...
+        RETURN_IF_ERROR(Expr::create_expr_trees(
+                _pool, broker_scan_node.delete_conjuncts, &_delete_expr_ctxs));
+    }
+    if (broker_scan_node.__isset.merge_type) {
+        _merge_type = broker_scan_node.merge_type;
+    } else {
+        _merge_type = TMergeType::APPEND;
+    }
+    if (_merge_type != TMergeType::MERGE && _delete_expr_ctxs.size() > 0) {
+        return Status::InvalidArgument("cannot support delete condition in APPEND or DELETE load.");
```
[the remark this answers is not in the document]
Comment (yangzhg) [reply be/src/exec/broker_scan_node.cpp:84]: delete on condition is for MERGE load,  APPEND is append all data in this load, DELETE is drop all data in this load, Merge is delete all data match delete condition and append others

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java [hunk 2]
```diff
@@ -280,7 +287,11 @@ protected void setOptional(CreateRoutineLoadStmt stmt) throws UserException {
         if (stmt.getMaxBatchSize() != -1) {
             this.maxBatchSizeBytes = stmt.getMaxBatchSize();
         }
+        if (stmt.getExecMemLimit() != -1) {
+            this.execMemLimit = stmt.getExecMemLimit();
```
[the remark this answers is not in the document]
Comment (yangzhg) [reply fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java:291]: execMemLimit in all load's property except routine load, add this just for keeping syntax consist.

File: fe/fe-core/pom.xml [hunk 3]
```diff
@@ -458,7 +458,7 @@ under the License.
                     <!--<cupDefinition>${doris.home}/gensrc/parser/sql_parser.cup</cupDefinition>-->
                     <className>SqlParser</className>
                     <symbolsName>SqlParserSymbols</symbolsName>
-                    <expectedConflicts>6</expectedConflicts>
+                    <expectedConflicts>7</expectedConflicts>
```
[the remark this answers is not in the document]
Comment (yangzhg) [reply fe/fe-core/pom.xml]: this will be 0 after the he

**Review by imay — CHANGES_REQUESTED** [review]
Should discuss about this feature.

**Pull request closed without merging**